Subject: [PATCH] 增加前台接口限流
---
Index: ruoyi-admin/src/main/java/com/ruoyi/web/controller/common/CommonController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/common/CommonController.java b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/common/CommonController.java
--- a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/common/CommonController.java	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/common/CommonController.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -1,5 +1,6 @@
 package com.ruoyi.web.controller.common;
 
+import com.ruoyi.common.annotation.RateLimiter;
 import com.ruoyi.common.config.RuoYiConfig;
 import com.ruoyi.common.constant.CacheConstants;
 import com.ruoyi.common.constant.Constants;
@@ -10,6 +11,7 @@
 import com.ruoyi.common.core.domain.entity.SysUser;
 import com.ruoyi.common.core.domain.model.LoginUser;
 import com.ruoyi.common.core.redis.RedisCache;
+import com.ruoyi.common.enums.LimitType;
 import com.ruoyi.common.utils.StringUtils;
 import com.ruoyi.common.utils.SysCache;
 import com.ruoyi.common.utils.file.FileUploadUtils;
@@ -231,6 +233,7 @@
     }
 
     @GetMapping("/sysInfo")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public RuoYiConfig sysInfo() {
         SysWebsite website = sysWebsiteService.getById(1);
         if (StringUtils.isNotBlank(website.getShortName())) {
Index: ruoyi-admin/src/main/java/com/ruoyi/web/controller/sale/SysSaleShopController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/sale/SysSaleShopController.java b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/sale/SysSaleShopController.java
--- a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/sale/SysSaleShopController.java	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/sale/SysSaleShopController.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -36,6 +36,7 @@
 import com.ruoyi.sale.service.ISysSaleOrderService;
 import com.ruoyi.sale.service.ISysSaleShopService;
 import com.ruoyi.system.domain.*;
+import com.ruoyi.system.domain.vo.CountVo;
 import com.ruoyi.system.mapper.*;
 import com.ruoyi.system.service.*;
 import com.ruoyi.web.controller.sale.vo.*;
@@ -101,22 +102,23 @@
      * 查询软件列表
      */
     @GetMapping("/appList")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public TableDataInfo appList(SysApp sysApp) {
         List<SaleAppVo> saleAppVoList = new ArrayList<>();
         List<SysApp> appList = sysAppMapper.selectSysAppList(sysApp);
+        Map<Long, CountVo> cardTemplateCountMap = sysCardTemplateMapper.selectSysCardTemplateOnSaleCountGroupByAppId();
+        Map<Long, CountVo> loginCodeTemplateCountMap = sysLoginCodeTemplateMapper.selectSysLoginCodeTemplateOnSaleCountGroupByAppId();
         for (SysApp app : appList) {
             if (app.getAuthType().equals(AuthType.ACCOUNT)) {
                 SysCardTemplate ct = new SysCardTemplate();
                 ct.setAppId(app.getAppId());
                 ct.setOnSale(UserConstants.YES);
-                saleAppVoList.add(new SaleAppVo(app.getAppId(), app.getAppName(), sysCardTemplateMapper.selectSysCardTemplateList(ct).size()));
+                saleAppVoList.add(new SaleAppVo(app.getAppId(), app.getAppName(), cardTemplateCountMap.getOrDefault(app.getAppId(), new CountVo()).getCount()));
             } else {
                 SysLoginCodeTemplate ct = new SysLoginCodeTemplate();
                 ct.setAppId(app.getAppId());
-                ct.setOnSale(UserConstants.YES);
-                saleAppVoList.add(new SaleAppVo(app.getAppId(), app.getAppName(), sysLoginCodeTemplateMapper.selectSysLoginCodeTemplateList(ct).size()));
+                saleAppVoList.add(new SaleAppVo(app.getAppId(), app.getAppName(), loginCodeTemplateCountMap.getOrDefault(app.getAppId(), new CountVo()).getCount()));
             }
-
         }
         return getDataTable(saleAppVoList);
     }
@@ -125,6 +127,7 @@
      * 查询卡密模板列表
      */
     @GetMapping("/listCategory")
+    @RateLimiter(limitType = LimitType.IP)
     public TableDataInfo cardTemplateList(Long appId) {
         SysApp app = sysAppMapper.selectSysAppByAppId(appId);
         List<SaleCardTemplateVo> saleCardTemplateVoList = new ArrayList<>();
@@ -167,6 +170,7 @@
     }
 
     @PostMapping("/checkStock")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult checkStock(@RequestBody SaleOrderVo saleOrderVo) {
 
         Payment payment = PaymentDefine.paymentMap.get(saleOrderVo.getPayMode());
@@ -207,7 +211,7 @@
     }
 
     @PostMapping("/createSaleOrder")
-    @RateLimiter(limitType = LimitType.IP)
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult createSaleOrder(@RequestBody SaleOrderVo saleOrderVo) {
 
 //        if (StringUtils.isAnyBlank(saleOrderVo.getContact(), saleOrderVo.getQueryPass())) {
@@ -295,7 +299,7 @@
     }
 
     @PostMapping("/createChargeOrder")
-    @RateLimiter(limitType = LimitType.IP)
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult createChargeOrder(@RequestBody ChargeOrderVo chargeOrderVo) {
         Payment payment = PaymentDefine.paymentMap.get(chargeOrderVo.getPayMode());
         if (payment == null) {
@@ -361,6 +365,7 @@
     }
 
     @GetMapping("/paySaleOrder")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult paySaleOrder(String orderNo) {
         if (orderNo == null) {
             throw new ServiceException("订单不存在", 400);
@@ -412,6 +417,7 @@
      * @return
      */
     @GetMapping("/fetchGoods")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult fetchGoods(String orderNo) {
         if (orderNo == null) {
             throw new ServiceException("订单不存在", 400);
@@ -430,6 +436,7 @@
     }
 
     @GetMapping("/getCardList")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult getCardList(@RequestParam("orderNo") String orderNo, @RequestParam(value = "queryPass", required = false) String queryPass) {
         if (orderNo == null) {
             throw new ServiceException("订单不存在", 400);
@@ -474,6 +481,7 @@
      * 查询销售订单列表，订单查询调用
      */
     @GetMapping("/querySaleOrderByContact")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public TableDataInfo querySaleOrderByContact(SysSaleOrder sysSaleOrder) {
         List<SysSaleOrder> list = sysSaleOrderService.selectSysSaleOrderQueryLimit5(sysSaleOrder);
         return getDataTable(list);
@@ -485,6 +493,7 @@
      * @return
      */
     @GetMapping("/getShopConfig")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult getShopConfig() {
         Map<String, Object> map = new HashMap<>();
         // 公告
Index: ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysAppVersionController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysAppVersionController.java b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysAppVersionController.java
--- a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysAppVersionController.java	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysAppVersionController.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -1,6 +1,7 @@
 package com.ruoyi.web.controller.system;
 
 import com.ruoyi.common.annotation.Log;
+import com.ruoyi.common.annotation.RateLimiter;
 import com.ruoyi.common.constant.CacheConstants;
 import com.ruoyi.common.core.controller.BaseController;
 import com.ruoyi.common.core.domain.AjaxResult;
@@ -8,6 +9,7 @@
 import com.ruoyi.common.core.page.TableDataInfo;
 import com.ruoyi.common.core.redis.RedisCache;
 import com.ruoyi.common.enums.BusinessType;
+import com.ruoyi.common.enums.LimitType;
 import com.ruoyi.system.domain.vo.ActivityMethodVo;
 import com.ruoyi.system.service.ISysAppService;
 import com.ruoyi.system.service.ISysAppVersionService;
@@ -149,6 +151,7 @@
      */
     @Log(title = "快速接入(扫码)", businessType = BusinessType.QUICK_ACCESS)
     @RequestMapping("/scan/{appVersionId}/{uuid}")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult getQuickAccessParams(@PathVariable Long appVersionId, @PathVariable String uuid) {
         String verifyKey = CacheConstants.CAPTCHA_UUID_KEY + uuid;
         if(redisCache.hasKey(verifyKey)) {
Index: ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysLoginController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysLoginController.java b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysLoginController.java
--- a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysLoginController.java	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysLoginController.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -6,6 +6,7 @@
 import com.ruoyi.common.core.domain.entity.SysMenu;
 import com.ruoyi.common.core.domain.entity.SysUser;
 import com.ruoyi.common.core.domain.model.LoginBody;
+import com.ruoyi.common.enums.LimitType;
 import com.ruoyi.common.utils.SecurityUtils;
 import com.ruoyi.framework.license.anno.LicenceCheck;
 import com.ruoyi.framework.web.service.SysLoginService;
@@ -39,8 +40,8 @@
      * @param vstr
      * @return
      */
-    @RateLimiter
     @GetMapping("/checkSafeEntrance")
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult checkSafeEntrance(@RequestParam("vstr") String vstr) {
         return AjaxResult.success().put("data", loginService.checkSafeEntrance(vstr));
     }
Index: ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysWebsiteController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysWebsiteController.java b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysWebsiteController.java
--- a/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysWebsiteController.java	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-admin/src/main/java/com/ruoyi/web/controller/system/SysWebsiteController.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -1,10 +1,12 @@
 package com.ruoyi.web.controller.system;
 
 import com.ruoyi.common.annotation.Log;
+import com.ruoyi.common.annotation.RateLimiter;
 import com.ruoyi.common.config.RuoYiConfig;
 import com.ruoyi.common.core.controller.BaseController;
 import com.ruoyi.common.core.domain.AjaxResult;
 import com.ruoyi.common.enums.BusinessType;
+import com.ruoyi.common.enums.LimitType;
 import com.ruoyi.common.exception.ServiceException;
 import com.ruoyi.common.utils.DateUtils;
 import com.ruoyi.common.utils.StringUtils;
@@ -36,6 +38,7 @@
      * 获取网站配置信息
      */
     @GetMapping
+    @RateLimiter(count = 10, limitType = LimitType.IP)
     public AjaxResult getConfig() {
 
         SysWebsite website = sysWebsiteService.getById(1);
Index: ruoyi-admin/src/main/resources/application.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-admin/src/main/resources/application.yml b/ruoyi-admin/src/main/resources/application.yml
--- a/ruoyi-admin/src/main/resources/application.yml	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-admin/src/main/resources/application.yml	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -60,8 +60,8 @@
 # 日志配置
 logging:
   level:
-    com.ruoyi: info
-    org.springframework: warn
+    com.ruoyi: debug #info
+    org.springframework: debug # warn
 
 # 用户配置
 user:
@@ -168,13 +168,13 @@
     lettuce:
       pool:
         # 连接池中的最小空闲连接
-        min-idle: 0
+        min-idle: 5
         # 连接池中的最大空闲连接
-        max-idle: 8
+        max-idle: 20
         # 连接池的最大数据库连接数
-        max-active: 8
+        max-active: 100
         # #连接池最大阻塞等待时间（使用负值表示没有限制）
-        max-wait: -1ms
+        max-wait: 2000
   jackson:
     date-format: yyyy-MM-dd HH:mm:ss
     time-zone: GMT+8
Index: ruoyi-system/src/main/java/com/ruoyi/system/domain/vo/CountVo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-system/src/main/java/com/ruoyi/system/domain/vo/CountVo.java b/ruoyi-system/src/main/java/com/ruoyi/system/domain/vo/CountVo.java
new file mode 100644
--- /dev/null	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
+++ b/ruoyi-system/src/main/java/com/ruoyi/system/domain/vo/CountVo.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -0,0 +1,13 @@
+package com.ruoyi.system.domain.vo;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+public class CountVo {
+
+    private long id;
+    private int count = 0;
+
+}
Index: ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysCardTemplateMapper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysCardTemplateMapper.java b/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysCardTemplateMapper.java
--- a/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysCardTemplateMapper.java	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysCardTemplateMapper.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -1,10 +1,13 @@
 package com.ruoyi.system.mapper;
 
 import com.ruoyi.system.domain.SysCardTemplate;
+import com.ruoyi.system.domain.vo.CountVo;
+import org.apache.ibatis.annotations.MapKey;
 import org.apache.ibatis.annotations.Param;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.Map;
 
 /**
  * 卡密模板Mapper接口
@@ -71,4 +74,11 @@
      * @return 卡密模板
      */
     public SysCardTemplate selectSysCardTemplateByAppIdAndTemplateName(@Param("appId") Long appId, @Param("templateName") String templateName);
+
+    /**
+     * 获取某APP下已上架的卡密模板数量
+     * @return
+     */
+    @MapKey("id")
+    public Map<Long, CountVo> selectSysCardTemplateOnSaleCountGroupByAppId();
 }
\ No newline at end of file
Index: ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysLoginCodeTemplateMapper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysLoginCodeTemplateMapper.java b/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysLoginCodeTemplateMapper.java
--- a/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysLoginCodeTemplateMapper.java	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-system/src/main/java/com/ruoyi/system/mapper/SysLoginCodeTemplateMapper.java	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -1,10 +1,13 @@
 package com.ruoyi.system.mapper;
 
 import com.ruoyi.system.domain.SysLoginCodeTemplate;
+import com.ruoyi.system.domain.vo.CountVo;
+import org.apache.ibatis.annotations.MapKey;
 import org.apache.ibatis.annotations.Param;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.Map;
 
 /**
  * 单码类别Mapper接口
@@ -30,6 +33,14 @@
      */
     public List<SysLoginCodeTemplate> selectSysLoginCodeTemplateList(SysLoginCodeTemplate sysLoginCodeTemplate);
 
+
+    /**
+     * 获取某APP下已上架的卡密模板数量
+     * @return
+     */
+    @MapKey("id")
+    public Map<Long, CountVo> selectSysLoginCodeTemplateOnSaleCountGroupByAppId();
+
     /**
      * 新增单码类别
      *
Index: ruoyi-system/src/main/resources/mapper/system/SysCardTemplateMapper.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-system/src/main/resources/mapper/system/SysCardTemplateMapper.xml b/ruoyi-system/src/main/resources/mapper/system/SysCardTemplateMapper.xml
--- a/ruoyi-system/src/main/resources/mapper/system/SysCardTemplateMapper.xml	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-system/src/main/resources/mapper/system/SysCardTemplateMapper.xml	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -249,4 +249,8 @@
         <include refid="selectSysCardTemplateVo"/>
         where ct.app_id = #{appId} and ct.card_name = #{templateName} and ct.del_flag = '0'
     </select>
+
+    <select id="selectSysCardTemplateOnSaleCountGroupByAppId" resultType="CountVo">
+        select app_id as id, count(1) as count from sys_card_template where on_sale = 'Y' group by app_id
+    </select>
 </mapper>
\ No newline at end of file
Index: ruoyi-system/src/main/resources/mapper/system/SysLoginCodeTemplateMapper.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-system/src/main/resources/mapper/system/SysLoginCodeTemplateMapper.xml b/ruoyi-system/src/main/resources/mapper/system/SysLoginCodeTemplateMapper.xml
--- a/ruoyi-system/src/main/resources/mapper/system/SysLoginCodeTemplateMapper.xml	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-system/src/main/resources/mapper/system/SysLoginCodeTemplateMapper.xml	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -111,6 +111,10 @@
         </if>
     </select>
 
+    <select id="selectSysLoginCodeTemplateOnSaleCountGroupByAppId" resultType="CountVo">
+        select app_id as id, count(1) as count from sys_login_code_template where on_sale = 'Y' group by app_id
+    </select>
+
     <select id="selectSysLoginCodeTemplateByTemplateId" parameterType="Long" resultMap="SysLoginCodeTemplateResult">
         <include refid="selectSysLoginCodeTemplateVo"/>
         where template_id = #{templateId}
Index: ruoyi-ui/package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruoyi-ui/package.json b/ruoyi-ui/package.json
--- a/ruoyi-ui/package.json	(revision b66722591c976db4fb9737158b5679108f525ee1)
+++ b/ruoyi-ui/package.json	(revision dde3d638f3da756cd69d68cfe4a62b4212d50fa9)
@@ -1,16 +1,16 @@
 {
   "name": "hywlyz",
-  "version": "1.8.0",
+  "version": "1.8.2",
   "ruoyi-version": "3.8.5",
   "description": "网络验证管理系统",
   "author": "coordsoft.com",
   "license": "MIT",
   "scripts": {
-    "dev": "vue-cli-service serve",
-    "build:prod": "vue-cli-service build",
-    "build:stage": "vue-cli-service build --mode staging",
-    "preview": "node build/index.js --preview",
-    "lint": "eslint --ext .js,.vue src"
+    "dev": "SET NODE_OPTIONS=--openssl-legacy-provider && vue-cli-service serve",
+    "build:prod": "SET NODE_OPTIONS=--openssl-legacy-provider &&  && vue-cli-service build",
+    "build:stage": "SET NODE_OPTIONS=--openssl-legacy-provider &&  && vue-cli-service build --mode staging",
+    "preview": "SET NODE_OPTIONS=--openssl-legacy-provider &&  && node build/index.js --preview",
+    "lint": "SET NODE_OPTIONS=--openssl-legacy-provider &&  && eslint --ext .js,.vue src"
   },
   "husky": {
     "hooks": {
